src/NzbDrone.Core.Test/HistoryTests/HistoryServiceFixture.cs

```diff
@@ -21,46 +22,93 @@ public class HistoryServiceFixture : CoreTest<HistoryService>
         private Profile _profileCustom;
+        private LanguageProfile _languageProfile;
 
         [SetUp]
         public void Setup()
         {
-            _profile = new Profile { Cutoff = Quality.WEBDL720p, Items = QualityFixture.GetDefaultQualities() };
-            _profileCustom = new Profile { Cutoff = Quality.WEBDL720p, Items = QualityFixture.GetDefaultQualities(Quality.DVD) };
+            var _languages = new List<ProfileLanguageItem>();
+            _languages.Add(new ProfileLanguageItem { Language = Language.English, Allowed = true });
+            _languages.Add(new ProfileLanguageItem { Language = Language.Spanish, Allowed = true });
+            _languages.Add(new ProfileLanguageItem { Language = Language.French, Allowed = true });
+
+
+            _profile = new Profile
+                {
+                    Cutoff = Quality.WEBDL720p,
+                    Items = QualityFixture.GetDefaultQualities(),
+                };
+
+            _profileCustom = new Profile
+                {
+                    Cutoff = Quality.WEBDL720p, 
+                    Items = QualityFixture.GetDefaultQualities(Quality.DVD),
+
+                };
+
+
+            _languageProfile = new LanguageProfile
+            {
+                Cutoff = Language.Spanish,
+                Languages = _languages
+            };
+        
         }
 
         [Test]
         public void should_return_null_if_no_history()
         {
             Mocker.GetMock<IHistoryRepository>()
-                .Setup(v => v.GetBestQualityInHistory(2))
-                .Returns(new List<QualityModel>());
+                .Setup(v => v.GetBestInHistory(2))
+                .Returns(new List<BestInHistory>());
 
-            var quality = Subject.GetBestQualityInHistory(_profile, 2);
+            var bestItem = Subject.GetBestInHistory(_profile, _languageProfile, 2);
 
-            quality.Should().BeNull();
+            bestItem.Should().BeNull();
         }
 
         [Test]
         public void should_return_best_quality()
         {
             Mocker.GetMock<IHistoryRepository>()
-                .Setup(v => v.GetBestQualityInHistory(2))
-                .Returns(new List<QualityModel> { new QualityModel(Quality.DVD), new QualityModel(Quality.Bluray1080p) });
+                .Setup(v => v.GetBestInHistory(2))
+                .Returns(new List<BestInHistory> 
+                { 
+                    new BestInHistory { Quality = new QualityModel(Quality.DVD), Language = Language.English }, 
+                    new BestInHistory{Quality = new QualityModel(Quality.Bluray1080p), Language = Language.English} 
+                });
 
-            var quality = Subject.GetBestQualityInHistory(_profile, 2);
+            var quality = Subject.GetBestInHistory(_profile, _languageProfile, 2).Quality;
 
             quality.Should().Be(new QualityModel(Quality.Bluray1080p));
         }
 
         [Test]
+        public void should_return_best_quality_with_language()
```

Comment:
I recently removed one use of the best quality in history method, I'll double check but it might be able to get killed off completely.